### ISSUE TITLE
Doc: Move `Class overview` section somewhere more prominent

### ISSUE THREAD

**Issue by LecrisUT** [teemtee/tmt#3077]
Particularly on [the page](https://tmt.readthedocs.io/en/stable/code/classes.html#class-overview) `Phases` and `Steps` parts should be made more visible so that users can more easily know:
- The steps order
- Which steps are guestless/guestful

I often get tripped up when I forget to add a `discover` step and I don't see anything running. Another example is from #3065 where it is easier to visualize where to put the container setup steps, if they know the step order and the concepts of guestless/guestful it helps intuit `provision` and `prepare` steps.

**Comment by happz** [comment]
I'd argue that this page is not supposed to be the primary source of information about steps for *end users*. The order of steps, the fact you need `discover` to discover tests, all the things you described are valid points, but more for the users while this particular page is for contributors. There is definitely an overlap between these two groups, but a small one. I'd prefer making improvements reflecting the missing key concepts on pages like https://tmt.readthedocs.io/en/stable/spec.html or https://tmt.readthedocs.io/en/stable/guide.html#the-first-steps.

I mean, "The Common class is the parent of most of the available classes, it provides common methods for logging, running commands and workdir handling.", that's hardly the way we should teach end users about the order of steps and that `prepare` runs things on a guest while `discover` runs things on user's workstation.

I believe @psss has some plans to improve the guide, introduction, first steps, refactoring the docs, so, FYI, @psss.

**Comment by LecrisUT** [comment]
> I'd argue that this page is not supposed to be the primary source of information about steps for end users

Sure sure. It's not navigable for a user, just using that to reference where those concepts are being documented. The main thing is just to move the concepts from there higher up.

**Comment by psss** [comment]
> I'd argue that this page is not supposed to be the primary source of information about steps for _end users_.

Yes, I'd keep the class overview focused on the `tmt` contributors. There is a plan to add a new section to the Guide:

* #737 

It should cover the list of steps, their order, selecting some of them. @LecrisUT, would that cover your main concern here?

**Comment by LecrisUT** [comment]
Yes, that could cover it. Just to make sure we also cover the terminology of Phase as well. Something like a diagram to show how an fmf file is translated to a list of steps and phases

**Comment by psss** [comment]
Ok, extended that issue description with this as well.
